<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8"/>
<meta content="width=device-width, initial-scale=1.0, user-scalable=yes" name="viewport"/>
<style>
/**
 * Hatchd Digital Mou Theme
 * @author: Hatchd
 * @version: 1.0.0
 */

/**
 * Webfonts
 */

@font-face {
  font-family: 'vistaslab-bold';
  src: url('http://hatchd.com.au/static/fonts/vistaslab/vistaslab-bold-071211001EmigreWebOnly.eot');
  src: url('http://hatchd.com.au/static/fonts/vistaslab/vistaslab-bold-071211001EmigreWebOnly.eot?#iefix') format('embedded-opentype'), url('http://hatchd.com.au/static/fonts/vistaslab/vistaslab-bold-071211001EmigreWebOnly.woff') format('woff');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'vistaslab';
  src: url('http://hatchd.com.au/static/fonts/vistaslab/vistaslab-light-071211001EmigreWebOnly.eot');
  src: url('http://hatchd.com.au/static/fonts/vistaslab/vistaslab-light-071211001EmigreWebOnly.eot?#iefix') format('embedded-opentype'), url('http://hatchd.com.au/static/fonts/vistaslab/vistaslab-light-071211001EmigreWebOnly.woff') format('woff');
  font-weight: normal;
  font-style: normal;
}

/**
 * Normalize
 * (trimmed down)
 */
    html {
      font-size: 100%;
    }

    a:active,
    a:hover {
      outline: 0;
    }


/**
 * Tables
 */
    table {
      border-collapse: collapse;
      border-spacing: 0;
    }

/**
 * Typography
 */

    body {
      font-size: 100%;
      line-height: 1.5em;
      font-family: 'vistaslab', serif;
      color: #000000;
    }

    h1,
    h2,
    strong,
    b{
      font-family: 'vistaslab-bold', serif;
    }

    p,
    li,
    dt,
    dd {
      font-size: 1em;
      line-height: 1.5em;
    }

    p {
      margin: 0 0 1em 0;
    }

    hr {
      display: block;
      clear: both;
      height: 1px;
      margin: 1em 0;
      border: none;
      background: #cccccc;
    }

    sup,
    sub {
      font-size: 75%;
      line-height: 0;
      position: relative;
      vertical-align: baseline;
    }

    sup {
      top: -0.5em;
    }

    sub {
      bottom: -0.25em;
    }

    ins {
      background-color: #cccccc;
    }

    del {
      text-decoration: line-through;
    }

    s {
      color: #cccccc;
      text-decoration: line-through;
    }

    mark {
      background-color: #cccccc;
      color: inherit;
      font-weight: normal;
    }

    small {
      font-weight: normal;
      font-style: normal;
    }

    strong,
    b {
      font-weight: normal;
    }

    em,
    i {
      font-style: italic;
    }

    dfn,
    abbr {
      font-style: inherit;
      position: relative;
    }

    .feature-text {
      color: #ef4136;
    }

    hgroup {
      position: relative;
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      line-height: 1.3em;
      font-weight: normal;
      margin: 1em 0 0.5em 0;
      position: relative;
      clear: both;
    }

    h1 {
      font-size: 1.875em;
    }

    h2 {
      font-size: 1.25em;
    }

    h3 {
      font-size: 1em;
      font-weight: bold;
    }

    ul,
    ol,
    dl {
      margin: 0 0 1em 2em;
      padding: 0;
    }

    ul li,
    ol li {
      margin: 0 0 0.25em 0;
    }

    ul {
      list-style: disc outside;
    }

    ul ul {
      margin: 0.25em 0 0.5em 1em;
    }

    ol {
      list-style: decimal outside;
    }

    ol ol {
      margin: 0.25em 0 0.5em 1.5em;
      list-style-type: lower-roman;
    }

    dl {
      margin-left: 0;
    }

    dt {
      font-weight: bold;
      margin: 0 0 0.5em 0;
      padding: 1em 0 0 0;
    }

    dd {
      margin: 0 0 0.5em 0;
    }

    blockquote {
      margin: 0 0 2em 2em;
      padding: 1em 0 0 0;
    }

    blockquote > p {
      padding: 0 1em;
      font-size: 1.6em;
      line-height: 1.2em;
    }

    blockquote > p:first-child:before,
    blockquote > p:first-child:after {
      content: '"';
    }

    q {
      font-style: italic;
      color: inherit;
    }

    cite {
      font-style: italic;
    }

    pre {
      white-space: pre;
      width: 100%;
      overflow: auto;
      margin: 2em 0 2em 0;
      box-sizing: border-box;
      -moz-box-sizing: border-box;
      -webkit-box-sizing: border-box;
    }

    code,
    samp,
    kbd {
      display: inline-block;
      padding: 0 0.5em;
    }

    var {
      font-style: italic;
      font-weight: bold;
    }

/**
 * Links
 */
    a,
    a:link,
    a:visited {
      color: #ef4136;
      text-decoration: underline;
    }

    a:hover {
      color: #ef4136;
      text-decoration: none;
    }

    a:active {
      color: #ef4136;
      text-decoration: underline;
    }

    ::selection {
      color: #ffffff;
      background: #cccccc;
    }

    ::-moz-selection {
      color: #ffffff;
      background: #cccccc;
    }


/**
 * Images
 */
    img {
      display: block;
      border: 0;
      margin: 0 0 1em 0;
      width: 100%;
    }

/**
 * Layout
 */
    body{
        max-width: 830px;
        width: 96%;
        margin:auto;
        padding:2em 0;
    }


</style>
<title>BENCHMARK</title>
<link href="css/toc.css" rel="stylesheet"/><script src="http://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script><script src="js/toc.min.js"></script></head>
<body>
<h1>BENCHMARK</h1>
<p>In the webpage, we provide information about benchmarks between SEQMINER, VariantAnnotation and GEMINI. Our benchmark was performed on Intel Xeon 5530 with 256 Gigabyte memory and enough hard drive space. This document provides necessary instructions in order to help the readers replicate our results.</p>
<p>We provide data and source codes. In general, we describe commonly used data sets (e.g. reference genome, gene definition files et al) in <a href="http:/zhanxw.github.io/seqminer/data.set.html">Data Set</a>, and we describe benchmark specific data sets and source codes in <a href="(*">Benchmark Directory</a>. After visiting <a href="(*">Benchmark Directory</a>, you will see a directory lists: <em>data</em> (benchmark specific data sets), <em>integration</em> (source codes where we benchmarked data integration), <em>extraction</em> (source codes where we benchmarked information retrieval).
We will introduce the details of each directories in subsequent sections.</p>
<h2>Data sets</h2>
<p>We provide all data sets or provide instructions on how to obtain them in this website. The data are categorized into two categories: (1) Common data sets (2) Benchmark specific data sets.</p>
<h3>Common data sets</h3>
<p>To download the following files or other commonly used data sets (e.g. reference genome, gene definition files et al), please refer to this description webpage <a href="*%20data.set.html">here</a>. The files listed below are frequently used during benchmarking.</p>
<ul>
<li>ALL.chr1.phase1_release_v3.20101123.snps_indels_svs.genotypes.vcf.gz</li>
</ul>
<p>Chromosome 1 genotype file from the 1000 Genome Project</p>
<ul>
<li>ALL.chr1.phase1_release_v3.20101123.snps_indels_svs.genotypes.vcf.gz.tbi</li>
</ul>
<p>Tabix index</p>
<ul>
<li>hg19_ljb_all.txt.gz</li>
</ul>
<p>dbNSFP database. It include Polyphen scores and SIFT scores.</p>
<ul>
<li>human.g1k.v37.fa</li>
</ul>
<p>Human reference genome build 37.</p>
<ul>
<li>human.g1k.v37.fa.fai</li>
</ul>
<p>Index file for the reference genome.</p>
<ul>
<li>knownGene.txt.gz</li>
</ul>
<p>Gene definition file.</p>
<h3>Benchmark specific data sets</h3>
<p>The following files can be downloaded directly.</p>
<ul>
<li>rand.chr1.range.100</li>
</ul>
<p>100 random selected genomic regions on chromosome 1</p>
<p>File link: <a href="http://www.zhanxw.com/seqminer/benchmark/data/rand.chr1.range.100">http://www.zhanxw.com/seqminer/benchmark/data/rand.chr1.range.100</a></p>
<ul>
<li>rand.chr1.gene.100</li>
</ul>
<p>100 random selected genes on chromosome 1</p>
<p>File link: <a href="http://www.zhanxw.com/seqminer/benchmark/data/rand.chr1.gene.100">http://www.zhanxw.com/seqminer/benchmark/data/rand.chr1.gene.100</a></p>
<h2>Data integration</h2>
<p>The directory <a href="http://www.zhanxw.com/seqminer/benchmark/integration/">integration</a> includes source codes for data integration. We provide relevant codes for SEQMINER, VariantAnnotation (va) and GEMINI.</p>
<p>In <em>seqminer</em> folder, we provide <em>annotate.R</em> source file. It demonstration how to use SEQMINER to annotate genetic variants. The command line is:</p>
<pre><code>Rscript annotation.R
</code></pre>
<p>In <em>va</em> folder, we provide the source code <em>annotate.R</em> . It uses VariantAnnotation to annotate variants in the chromosome 1 genotype file of VCF format. Ideally, VariantAnnotation performs best when it can annotate all variants in one batch. However, due to our memory limit (256G memory), we let VariantAnnotation annotate 20,000 variants per batch. The following command line was from the provided  <em>cmd.sh</em> file:</p>
<pre><code>Rscript annotate.R ALL.chr1.phase1_release_v3.20101123.snps_indels_svs.genotypes.vcf.gz anno.chr1.chunk20000.vcf 20000
</code></pre>
<p>NOTE: the last parameter <code>20000</code> specifies 20,000 variants per batch.
In our computational environment, we have also tried 5,000 or 10,000 variant per batch and found the performance of VariantAnnotation degraded. We also noted that specifying a larger number than 20,000 will break VariantAnnotation due to memory limitation.</p>
<p>In <em>gemini</em> folder, we built GEMINI database projects and import annotation information. The annotation was performed by SnpEff with success and VEP without success. After annotation, we used the following command to import compressed VCF file (see <em>import.sh</em>):</p>
<pre><code>gemini load -v snpEff.annotated.wgs.vcf.gz -t snpEff --cores 8 snpEff.annotated.wgs.db
</code></pre>
<h2>Information retrieval</h2>
<p>Benchmarking of the information retrieval step was provided in the <em>extract</em> folder.</p>
<p>We benchmarked each software (SEQMINER and VariantAnnotation) in two settings:</p>
<p>(1) extract nonsynonymous variants from <strong>100 regions</strong>;</p>
<p>(2) extract nonsynonymous variants from <strong>100 genes</strong>.</p>
<p>These two settings correspond to two directories, “<em>chr1.nonsyn.gene</em>” and “<em>chr1.nonsyn.range</em>”.</p>
<p>Similar to the <em>Data Integration</em> section, we provide source codes under each directory.</p>
<p>In <em>seqminer</em> folder, to replicate our benchmark results in the Table 1 in our SEQMINER manuscript in the first setting (extracting non synonymous variants from 100 regions), please download the script at:</p>
<p><a href="http://www.zhanxw.com/seqminer/benchmark/extraction/seqminer/chr1.nonsyn.range/script.oneByone.R">http://www.zhanxw.com/seqminer/benchmark/extraction/seqminer/chr1.nonsyn.range/script.oneByone.R</a></p>
<p>to your local <em>seqminer/chr1.nonsyn.range</em> directory.</p>
<p>Make sure the <code>vcfFile</code> variable in the source codes exists, and then run the following command to extract 1000 regions:</p>
<pre><code>Rscript script.oneByone.R
</code></pre>
<p>In <code>va</code> folder, its directory structure mimic the <code>seqminer</code> settings. To extract 100 randomly selected region, you will change current directory to <em>va/chr1.nonsyn.range</em>, and execute <code>Rscript script.oneByone.R</code>.</p>
<h2>Contact</h2>
<p>Please contact Xiaowei Zhan <a href="mailto:zhanxw@gmail.com">zhanxw@gmail.com</a> or Dajiang Liu <a href="mailto:dajiang.liu@outlook.com">dajiang.liu@outlook.com</a> for comments or suggestions.</p>
<p>Last update: November 13, 2014</p>
<script type="text/javascript">
   var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-21871925-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script><div id="toc"></div><script type="text/javascript">$('#toc').toc();</script></body>
</html>

